### PR TITLE
Add multipart/form-data JSON part support

### PR DESCRIPTION
**Pull request by bluk**
* Add GIF mimetype support
* Add `FormPart` to allow serialization of individual parts with custom headers
* Change `Encoder` to take a list of serializers
* Change the order to ensure that file parts are last

### CODE REVIEW

File: tests/encoder_test.py [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 from braintreehttp import File
 from braintreehttp.encoder import Encoder
+from braintreehttp.serializers import FormPart
```
Comment (sdcoffey) [inline tests/encoder_test.py]: Probably can bump this down onto the next import statement

File: tests/__init__.py [hunk 2]
```diff
@@ -2,3 +2,4 @@
 from tests.http_response_test import *
 from tests.encoder_test import *
 from tests.file_test import *
+from tests.serializers.form_part_test import *
```
Comment (sdcoffey) [inline tests/__init__.py]: This line is causing problems [in travis](https://travis-ci.org/braintree/braintreehttp_python/jobs/359607711)

**Review by sdcoffey — CHANGES_REQUESTED** [review]
Looks good, just gotta fix the travis build.

**Comment by bluk** [comment]
@sdcoffey Can you please take another pass? Thanks.

**Review by sdcoffey — APPROVED** [review]
Looks good!

**Pull request merged**